File: Plugins/Aspose_Slides_Java_for_PHP/src/aspose/slides/WorkingWithPresentation/Properties.php
<?php

namespace Aspose\Slides\WorkingWithPresentation;

use com\aspose\slides\Presentation as Presentation;
use com\aspose\slides\SaveFormat as SaveFormat;


class Properties {

    public static function run($dataDir=null)
    {
        # Accessing Built-in Properties
        Properties::get_properties($dataDir);

        # Modifying Built-in Properties
        Properties::update_properties($dataDir);

        # Adding Custom Document Properties
        Properties::add_custom_properties($dataDir);

        # Removing Document Properties
        //Properties::remove_property($dataDir);
    }

    public static function get_properties($dataDir=null)
    {

        # Instantiate the Presentation class that represents the presentation
        $pres = new Presentation($dataDir . "HelloWorld.pptx");

        # Create a reference to IDocumentProperties object associated with Presentation
        $dp = $pres->getDocumentProperties();

        # Display the builtin properties
        print "Category : " . $dp->getCategory() . PHP_EOL;
        print "Current Status : " . $dp->getContentStatus() . PHP_EOL;
        print "Creation Date : " . $dp->getCreatedTime() . PHP_EOL;
        print "Author : " . $dp->getAuthor() . PHP_EOL;
        print "Description : " . $dp->getComments() . PHP_EOL;
        print "KeyWords : " . $dp->getKeywords() . PHP_EOL;
        print "Last Modified By : " . $dp->getLastSavedBy() . PHP_EOL;
        print "Supervisor : " . $dp->getManager() . PHP_EOL;
        print "Modified Date : " . $dp->getLastSavedTime() . PHP_EOL;
        print "Presentation Format : " . $dp->getPresentationFormat() . PHP_EOL;
        print "Last Print Date : " . $dp->getLastPrinted() . PHP_EOL;
        print "Is Shared between producers : " . $dp->getSharedDoc() . PHP_EOL;
        print "Subject : " . $dp->getSubject() . PHP_EOL;
        print "Title : " . $dp->getTitle() . PHP_EOL;

    }

    public static function update_properties($dataDir=null)
    {

        # Instantiate the Presentation class that represents the presentation
        $pres = new Presentation($dataDir . "HelloWorld.pptx");

        # Create a reference to IDocumentProperties object associated with Presentation
        $dp = $pres->getDocumentProperties();

        # Set the builtin properties
        $dp->setAuthor ("Aspose.Slides for Java");
        $dp->setTitle ("Modifying Presentation Properties");
        $dp->setSubject ( "Aspose Subject");
        $dp->setComments ( "Aspose Description");
        $dp->setManager ( "Aspose Manager");

        # Save your presentation to a file
        $save_format = new SaveFormat();
        $pres->save($dataDir . "DocProps.pptx", $save_format->Pptx);

        print "Properties have been updated, Please check output file.";

    }

    public static function add_custom_properties($dataDir=null)
    {

        # Instantiate the Presentation class that represents the presentation
        $pres = new Presentation($dataDir . "HelloWorld.pptx");

        # Getting Document Properties
        $dp = $pres->getDocumentProperties();

        # Adding Custom properties
        $dp->set_Item("New Custom" , 12);
        $dp->set_Item("My Name","Mudassir");
        $dp->set_Item("Custom", 124);

        # Saving presentation
        $save_format = new SaveFormat();
        $pres->save($dataDir . "CustomDemo.pptx", $save_format->Pptx);

        print "Added custom properties, please check output file.";

    }

    public static function remove_property($dataDir=null)
    {
        # Instantiate the Presentation class that represents the presentation
        $pres = new Presentation($dataDir . "HelloWorld.pptx");

        # Getting Document Properties
        $dp = $pres->getDocumentProperties();

        # Getting property name at particular index
        $property_name = $dp->getPropertyName(2);

        # Removing selected property
        $dp->remove($property_name);

        # Saving presentation
        $save_format = new SaveFormat();
        $pres->save($dataDir . "RemoveDP.pptx", $save_format->Pptx);

        print "Remove document property, please check output file.";
    }

}

File: Plugins/Aspose-Slides-Java-for-Jython/asposeslides/WorkingWithPresentation/Properties.py
from asposeslides import Settings
from com.aspose.slides import Presentation
from com.aspose.slides import SaveFormat
from com.aspose.slides import XpsOptions

class Properties:

    def __init__(self):
                        
         
        # Accessing Built-in Properties
        self.get_properties()

        # Modifying Built-in Properties
        self.update_properties()

        # Adding Custom Document Properties
        self.add_custom_properties()

        # Removing Document Properties
        self.remove_property()
        
    def get_properties(dataDir):
    
        dataDir = Settings.dataDir + 'WorkingWithPresentation/Properties/'
        
         # Instantiate the Presentation class that represents the presentation
        pres = Presentation(dataDir + "HelloWorld.pptx")

        # Create a reference to IDocumentProperties object associated with Presentation
        dp = pres.getDocumentProperties()

        # Display the builtin properties
        print "Category : " + dp.getCategory()
        print "Current Status : " + dp.getContentStatus()
        print "Creation Date : "
        dp.getCreatedTime()
        print "Author : " + dp.getAuthor()
        print "Description : " + dp.getComments()
        print "KeyWords : " + dp.getKeywords()
        print "Last Modified By : " + dp.getLastSavedBy()
        print "Supervisor : " + dp.getManager()
        print "Modified Date : " 
        dp.getLastSavedTime()
        print "Presentation Format : "
        dp.getPresentationFormat()
        print "Last Print Date : " 
        dp.getLastPrinted()
        print "Is Shared between producers : "
        dp.getSharedDoc()
        print "Subject : "
        dp.getSubject()
        print "Title : "
        dp.getTitle()
        
    def update_properties(dataDir):
        
        dataDir = Settings.dataDir + 'WorkingWithPresentation/Properties/'
        
        # Instantiate the Presentation class that represents the presentation
        pres = Presentation(dataDir + "HelloWorld.pptx")

        # Create a reference to IDocumentProperties object associated with Presentation
        dp = pres.getDocumentProperties()

        # Set the builtin properties
        dp.setAuthor ("Aspose.Slides for Java")
        dp.setTitle ("Modifying Presentation Properties")
        dp.setSubject ( "Aspose Subject")
        dp.setComments ( "Aspose Description")
        dp.setManager ( "Aspose Manager")

        # Save your presentation to a file
        save_format = SaveFormat
        pres.save(dataDir + "DocProps.pptx", save_format.Pptx)

        print "Properties have been updated, Please check output file."
        
    def add_custom_properties(dataDir):
        
        dataDir = Settings.dataDir + 'WorkingWithPresentation/Properties/'
        
        #Instantiate the Presentation class that represents the presentation
        pres = Presentation(dataDir + "HelloWorld.pptx")

        # Getting Document Properties
        dp = pres.getDocumentProperties()

        # Adding Custom properties
        dp.set_Item("New Custom" , 12)
        dp.set_Item("My Name","Mudassir")
        dp.set_Item("Custom", 124)

        # Saving presentation
        save_format = SaveFormat
        pres.save(dataDir + "CustomDemo.pptx", save_format.Pptx)

        print "Added custom properties, please check output file."
        
    
    def remove_property(dataDir):
        
        dataDir = Settings.dataDir + 'WorkingWithPresentation/Properties/'
        
        # Instantiate the Presentation class that represents the presentation
        pres = Presentation(dataDir + "HelloWorld.pptx")

        # Getting Document Properties
        dp = pres.getDocumentProperties()

        # Getting property name at particular index
        property_name = dp.getPropertyName(2)

        # Removing selected property
        dp.remove(property_name)

        # Saving presentation
        save_format = SaveFormat
        pres.save(dataDir + "RemoveDP.pptx", save_format.Pptx)

        print "Remove document property, please check output file."

if __name__ == '__main__':        
    Properties()